samples/subsys/usb/hid-mouse/src/main.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2018 qianfan Zhao
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <board.h>
+#include <device.h>
+#include <gpio.h>
+
+#include <usb/usb_device.h>
+#include <usb/class/usb_hid.h>
+
+/* change this to use another GPIO port */
+#ifdef SW0_GPIO_NAME
+#define PORT	SW0_GPIO_NAME
+#else
+#error SW0_GPIO_NAME needs to be set in board.h
+#endif
+
+/* change this to use another GPIO pin */
+#ifdef SW0_GPIO_PIN
+#define PIN     SW0_GPIO_PIN
+#else
+#error SW0_GPIO_PIN needs to be set in board.h
+#endif
+
+static const u8_t hid_report_desc[] = {
```

Comment:
What is a `hid_report_desc`, and what are the possible values?

If I wanted to develop a HID device, I need to know what are the possible values for this. Can I find this information in the USB specification documents? In some other webpage? 

Adding a link here to the relevant information would be good IMHO.

Reply:
That is too much for sample, absolutely not needed here.
PS:
http://www.usb.org/developers/hidpage/
http://www.usb.org/developers/hidpage/Hut1_12v2.pdf

Reply:
and USB HID 1.1 spec: http://www.usb.org/developers/hidpage/HID1_11.pdf

there has lost of examples in Hut1_12v2.pdf, also you can found more example by capture your usb mouse, keyboard, game pad and so on.

Reply:
And what about adding that link at `samples/subsys/usb/hid-mouse/README.rst`? I think that would make it easier for developers new to USB HID to find their way.

Reply:
This maybe should go to USB HID stack documentation, not to the sample. We already have sample without this information.